nrf_modem/doc/full-dfu.rst

```diff
@@ -0,0 +1,113 @@
+Full Modem firmware update
+##########################
+
+The modem library supports updating the full modem firmware when initialized in Full DFU mode via the ``nrf_modem_full_dfu.h`` interface.
+When the library is initialized in Full DFU mode, the only operation that can be carried out is a full modem firmware update.
+In particular, sockets are not available to the application.
+
+It is possible to switch between Full DFU and Normal operation modes by re-initializing the library.
+
+Memory Requirements
+*******************
+
+The total size of all shared memory regions must be at least 8220 bytes.
+@note:
+When using NCS, library initialization is done via the glue, which will configure the library shared memory regions' size and position in RAM.
+Use the Kconfig options belonging to the glue to ensure that the total size of the library shared memory region is as large as necessary.
+
+Modem library initialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to update the full modem firmware, the library must be initialized in `FULL_DFU` mode as shown below::
+
+	/* Shutdown modem to prepare for DFU */
+	nrf_modem_init(params, FULL_DFU_MODE);
+
+If the library has already been initialized in `NORMAL_MODE` it must be shut down via `nrf_modem_shutdown` and re-initialized as shown below::
+
+	nrf_modem_init(params, NORMAL_MODE);
+	/* Shutdown modem to prepare for DFU */
+	nrf_modem_shutdown();
+	nrf_modem_init(params, FULL_DFU_MODE);
+
+Considerations for corrupted modem firmware
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When designing your application, you might also consider whether you must
+tolerate interrupted DFU process, which leads to corrupted modem firmware.
+There might be various reasons, for example out of power situation, that might
+interrupt the update. Modem does not have back-up ROM and therefore interrupting
+the DFU leads to situation where modem does not boot.
+
+To workaround this situation, either application must tolerate the situation,
+or another means of recovery is provided, for example re-flashing the modem with
+PC tools.
+
+And then initialize modem library manually::
```

Comment:
The double colons are visible in the rendered output, it looks a bit strange. Is on purpose?

Reply:
Fixed